### ISSUE TITLE
PLC1802 has false negatives for `ascii` and `sorted`

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#14690]
[`len-test` (PLC1802)](https://docs.astral.sh/ruff/rules/len-test/) in Ruff 0.8.1 does not recognize the builtins [`ascii`](https://docs.python.org/3/library/functions.html#ascii) and [`sorted`](https://docs.python.org/3/library/functions.html#sorted) as returning sequences, leading to false negatives.

```console
$ cat plc1802.py
bool(len(ascii(1)))
bool(len(sorted("")))

$ ruff check --preview --isolated --select PLC1802 plc1802.py --output-format concise
All checks passed!
```

Compare similar builtins with true positives.
```console
$ cat plc1802.py 
bool(len(repr(1)))
bool(len(list("")))

$ ruff check --preview --isolated --select PLC1802 plc1802.py --output-format concise
plc1802.py:1:6: PLC1802 [*] `len(repr(1))` used as condition without comparison
plc1802.py:2:6: PLC1802 [*] `len(list(""))` used as condition without comparison
Found 2 errors.
[*] 2 fixable with the `--fix` option.
```
